LoopKit/CarbKit/CarbStore.swift

```diff
@@ -215,46 +217,62 @@ public final class CarbStore: HealthKitSampleStore {
 
         cacheStore.onReady { (error) in
             guard error == nil else { return }
+            
+            cacheStore.fetchAnchor(key: CarbStore.queryAnchorMetadataKey) { (anchor) in
+                self.queue.async {
+                    self.queryAnchor = anchor
+            
+                    if !self.authorizationRequired {
+                        self.createQuery()
+                    }
 
-            // Migrate modifiedCarbEntries and deletedCarbEntryIDs
-            self.cacheStore.managedObjectContext.perform {
-                for entry in UserDefaults.standard.modifiedCarbEntries ?? [] {
-                    let object = CachedCarbObject(context: self.cacheStore.managedObjectContext)
-                    object.update(from: entry)
-                }
+                    // Migrate modifiedCarbEntries and deletedCarbEntryIDs
+                    self.cacheStore.managedObjectContext.perform {
+                        for entry in UserDefaults.standard.modifiedCarbEntries ?? [] {
+                            let object = CachedCarbObject(context: self.cacheStore.managedObjectContext)
+                            object.update(from: entry)
+                        }
 
 
-                for externalID in UserDefaults.standard.deletedCarbEntryIds ?? [] {
-                    let object = DeletedCarbObject(context: self.cacheStore.managedObjectContext)
-                    object.externalID = externalID
-                }
+                        for externalID in UserDefaults.standard.deletedCarbEntryIds ?? [] {
+                            let object = DeletedCarbObject(context: self.cacheStore.managedObjectContext)
+                            object.externalID = externalID
+                        }
 
-                self.cacheStore.save()
-            }
+                        self.cacheStore.save()
+                    }
+                    
 
-            UserDefaults.standard.purgeLegacyCarbEntryKeys()
+                    UserDefaults.standard.purgeLegacyCarbEntryKeys()
             
-            // Carb model settings based on the selected absorption model
-            switch self.carbAbsorptionModel {
-            case .linear:
-                self.settings = CarbModelSettings(absorptionModel: LinearAbsorption(), initialAbsorptionTimeOverrun: absorptionTimeOverrun, adaptiveAbsorptionRateEnabled: false)
-            case .nonlinear:
-                self.settings = CarbModelSettings(absorptionModel: PiecewiseLinearAbsorption(), initialAbsorptionTimeOverrun: absorptionTimeOverrun, adaptiveAbsorptionRateEnabled: false)
-            case .adaptiveRateNonlinear:
-                self.settings = CarbModelSettings(absorptionModel: PiecewiseLinearAbsorption(), initialAbsorptionTimeOverrun: 1.0, adaptiveAbsorptionRateEnabled: true, adaptiveRateStandbyIntervalFraction: 0.2)
+                    // Carb model settings based on the selected absorption model
+                    switch self.carbAbsorptionModel {
```

Comment:
[nit] I don't understand why the settings are only configured *after* the store is ready. Not particularly important, but I don't see why this isn't done before `carbStore.onReady`. Or is `settings` used as a "gatekeeper" to prevent/fail before store is ready?